test/e2e/framework/prometheus.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2022 The Cluster Monitoring Operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
+
+	routev1 "github.com/openshift/api/route/v1"
+)
+
+func (f *Framework) MakeBasicPrometheus(ns, name, group string, replicas int32, secrets []string) *monitoringv1.Prometheus {
```

Comment:
Since we're only using this function to setup a remote write receiver, we could merge it with `AddRemoteReceiveWithWebTLSToPrometheus`?

```suggestion
func (f *Framework) MakeRemoteReceiverPrometheus(name string, tlsSecretName string) *monitoringv1.Prometheus {
```